spec/models/miq_server/log_management_spec.rb

```diff
@@ -46,28 +127,89 @@
+        expect(@miq_server.log_depot("Zone")).to eq(zone_depot)
       end
 
       it "server and zone log_file_depot configured" do
-        server_depot = FactoryGirl.create(:file_depot)
-        zone_depot   = FactoryGirl.create(:file_depot)
         @miq_server.log_file_depot = server_depot
         @zone.log_file_depot = zone_depot
+        expect(@miq_server.log_depot("Zone")).to eq(zone_depot)
+        expect(@miq_server.log_depot("MiqServer")).to eq(server_depot)
+      end
+    end
+
+    describe "#post_historical_logs" do
+      context "Server" do
+        include_examples "post_[type_of_log]_logs", "MiqServer", :historical
+      end
+
+      context "Zone" do
+        include_examples "post_[type_of_log]_logs", "Zone", :historical
+      end
+    end
+
+    describe "#post_current_logs" do
+      context "Server" do
+        include_examples "post_[type_of_log]_logs", "MiqServer", :current
+      end
+
+      context "Zone" do
+        include_examples "post_[type_of_log]_logs", "Zone", :current
+      end
+    end
+    def post_logs(options)
+      taskid = options[:taskid]
+      task = MiqTask.find(taskid)
+      context_log_depot = log_depot(options[:context])
+
+      # the current queue item and task must be errored out on exceptions so re-raise any caught errors
+      raise "Log depot settings not configured" unless context_log_depot
+      context_log_depot.update_attributes(:support_case => options[:support_case].presence)
+
+      post_historical_logs(taskid, context_log_depot) unless options[:only_current]
+      post_current_logs(taskid, context_log_depot)
+      task.update_status("Finished", "Ok", "Log files were successfully collected")
+    end
+
+    describe "#post_logs" do
+      context "Zone collection log requested, Zone depot is defined, MiqServer is defined" do
+        include_examples "post_logs_uses_depot", true, true, "Zone"
+      end
+
+      context "Zone collection log requested, Zone depot is defined, MiqServer is not defined" do
+        include_examples "post_logs_uses_depot", true, false, "Zone"
+      end
+
+      context "Zone collection log requested, zone depot is not defined, MiqServer defined" do
+        include_examples "post_logs_fails", false, true, "Zone"
+      end
+
+      context "Zone collection log requested, zone depot is not defined, MiqServer is not defined" do
+        include_examples "post_logs_fails", false, false, "Zone"
+      end
+
+      context "MiqServer collection log requested, Zone depot is defined, MiqServer is defined" do
+        include_examples "post_logs_uses_depot", true, true, "MiqServer"
+      end
+
+      context "MiqServer collection log requested, Zone depot is defined, MiqServer is not defined" do
+        include_examples "post_logs_fails", true, false, "MiqServer"
+      end
+
+      context "MiqServer collection log requested, zone depot is not defined, server defined" do
+        include_examples "post_logs_uses_depot", false, true, "MiqServer"
```

Comment:
These are a bit hard to follow.  I can't figure out what false and true mean at a glance.  Maybe it's just me.  @bdunne ?